### ISSUE TITLE
FlxSound: Get mp3 ID3 tag from streaming sounds

### ISSUE THREAD

**Issue by SeiferTim** [AdamAtomic/flixel#43]
Minor suggestion: 

Change logic in FlxSound to do something like:

<pre>_sound = new Sound();
_sound.addEventListener(Event.ID3, id3Get);
_sound.load(new URLRequest(SoundURL));</pre>

and add a new function:

<pre>private function id3Get(event:Event):void
{
SongName = _s.id3.songName;
HasName = true;
}</pre>

The purpose of "HasName" is that it might take a few ms to get the songname after streaming is started, so in a loop (like in update), just keep checking if (HasName) FlxG.log(SongName); or something.

**Comment by AdamAtomic** [comment]
ok new FlxSound variables playing, name and artist will be available in 2.23 :)

**Comment by snotwadd20** [comment]
Hey Adam,

```
I'm playing around with the dev version (really cool, BTW), and I ran into a bug with this code. If you try to stream in an MP3 file that doesn't have ID3 information for Artist or Song Name, the program will crash. I fixed it with this: 
```

protected function gotID3(event:Event=null):void
        {
            FlxG.log("got ID3 info!");
            if(_sound.id3.songName != null && _sound.id3.songName.length > 0)
                name = _sound.id3.songName;
            if(_sound.id3.artist != null && _sound.id3.artist.length > 0)
                artist = _sound.id3.artist;
            _sound.removeEventListener(Event.ID3, gotID3);
        }

EDIT: I'm gonna open up an issue, just in case you can't track comments.
